### PR TITLE
Gas price changes and subprovider

### PR DESCRIPTION
**Pull request by mikeshultz**
### Description:

This is quick and dirty, duplicates code, and should be cleaned up later.  Trying to port ZeroClientProvider to our web3-provider will take more effort.  

This adds `EthGasStationProvider` to `mobile-bridge` and adjusts gas price to average across the board.

Here's the Rinkeby test build: https://originprotocol.github.io/test-builds/gas-provider

### Checklist:

- [x] Test your work and double-check to confirm that you didn't break anything
- [ ] [Wrap any new text/strings with `fbt`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#wrapping-text) for translation
- [ ] If there are any new/changed strings to translate, [`npm run translate`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#integrating-translations)
- [ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/docs)

### CODE REVIEW

**Comment by nick** [comment]
Seems OK... haven't tested it though

**Comment by mikeshultz** [comment]
### DO NOT MERGE YET

**Comment by mikeshultz** [comment]
@nick @franckc, do you think this should be enabled for desktop dapp?  I briefly tested it earlier and it worked okay with MM, but I'm a little worried about unintended consequences of adding `@origin/web3-provider` in such a short time without testing.

**Comment by mikeshultz** [comment]
ok, well, on the urging from @micahalcorn, I've enabled `@origin/web3-provider` across the board, including `EthGasStationProvider`.  Test please.

File: packages/web3-provider/src/subproviders/EthGasStationProvider.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ class EthGasStationProvider extends SubProvider {
     const jason = await res.json()
     if (typeof jason[GAS_PRICE_KEY] !== 'undefined') {
       // values come from EGS as tenths of gwei
-      return String(jason[GAS_PRICE_KEY] * 1e8)
+      return '0x' + (jason[GAS_PRICE_KEY] * 1e8).toString(16)
```
Comment (franckc) [inline packages/web3-provider/src/subproviders/EthGasStationProvider.js]: That won't break the relayer, will it ?

Comment (mikeshultz) [reply packages/web3-provider/src/subproviders/EthGasStationProvider.js]: tests passed

Comment (mikeshultz) [reply packages/web3-provider/src/subproviders/EthGasStationProvider.js]: I feel like I should elaborate that standad JSON-RPC interfaces return in hex.  This shouldn't break anything, tests passed, but it is a valid concern and we should test it when it gets deployed.

**Pull request merged**